loopy/transform/batch.py

```diff
@@ -73,6 +73,9 @@ def map_variable(self, expr, expn_state):
         if not self.needs_batch_subscript(expr.name):
             return super(_BatchVariableChanger, self).map_variable(expr, expn_state)
 
+        from pymbolic.primitives import Subscript
+        if not isinstance(expr, Subscript):
+            return Subscript(expr, self.batch_iname_expr)
         return expr.aggregate[self.batch_iname_expr]
```

Comment:
Whoops. Sorry for not spotting this last night--this is just broken. In `map_variable`, it's guaranteed that `expr` is a `Variable`, and `Variable`s don't have `.aggregate`.

Fixed (I think): e779b24